### PR TITLE
[5.0.0-rc2] Backport Fix chrome/selenium tests (#25330)

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Back port of https://github.com/dotnet/aspnetcore/pull/25330 to rc2. Since it's only testing, I'm going to mark this as tell mode.

### CODE REVIEW

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Approved but it appears different tests are failing on this branch

**Comment by JunTaoLuo** [comment]
I see one rc2 test that's affected by browser logs which I can disable. There's one blazor pwa template I'm not sure about and the rest look like setting cultures is broken. Not sure why those would be broken.

**Comment by JunTaoLuo** [comment]
Taking a quick look, it does seem like there are some test issues, at least for the culture-related ones.

For example:
```
Assert.Equal() Failure
                      ↓ (pos 12)
Expected: 02.09.2020 0:00:00
Actual:   02.09.2020 00:00:00
                      ↑ (pos 12)

Assert.Equal() Failure
              ↓ (pos 4)
Expected: 2020-09-02 오전 12:00:00
Actual:   2020. 9. 2. 오전 12:00:00
              ↑ (pos 4)
```

File: src/Components/test/testassets/TestServer/ServerStartup.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public ServerStartup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc();
-            services.AddServerSideBlazor(options => options.DetailedErrors = true);
+            services.AddServerSideBlazor();
```
Comment (SteveSandersonMS) [inline src/Components/test/testassets/TestServer/ServerStartup.cs:22]: Out of interest, why is this change required?

**Review by SteveSandersonMS — APPROVED** [review]
Changes look good to me. Thanks for pursuing this, @JunTaoLuo!

As for why certain culture-specific tests are failing, I'm unsure. @pranavkm, you recently did some work to set up culture-specific E2E tests. Do you have any insights into what's going on here?

File: src/Components/test/testassets/TestServer/ServerStartup.cs (hunk 1 above)
Comment (JunTaoLuo) [reply src/Components/test/testassets/TestServer/ServerStartup.cs:22]: This option controls how some exceptions are logged: https://github.com/dotnet/aspnetcore/blob/ec2c7b478769c68870ec860592be27cc8920519b/src/Components/Server/src/Circuits/RemoteJSRuntime.cs#L43-L56
and it turns out some tests check for this behaviour: https://github.com/dotnet/aspnetcore/blob/9a1810c1dbe432fc7bc7e8bc68fa22ab787c0452/src/Components/test/E2ETest/ServerExecutionTests/ServerInteropTestDefaultExceptionsBehavior.cs#L66-L68

given there were several tests that seemed to be dependent on this options, I just removed it. It's possible that you might consider changing the log messages in the tests instead.

Comment (SteveSandersonMS) [reply src/Components/test/testassets/TestServer/ServerStartup.cs:22]: This option affects the exception messages that are returned to the browser in JS interop calls. It doesn't only affect what gets logged to the browser console, it affects the exceptions that can be caught by JS code. The test code looks at the exceptions caught by JS code.

So I'm unclear on why this needs to be changed. I thought the only issue was with reading the browser console logs with the new version of Selenium, but the test in question (`DotNetExceptionDetailsAreNotLoggedByDefault`) wasn't doing that anyway.

> It's possible that you might consider changing the log messages in the tests instead.

Could you clarify what you're suggesting?

**Review by SteveSandersonMS — COMMENTED** [review]
Seeking clarification on comment above

File: src/Components/test/testassets/TestServer/ServerStartup.cs (hunk 1 above)
Comment (JunTaoLuo) [reply src/Components/test/testassets/TestServer/ServerStartup.cs:22]: No this has nothing to do with the browser console logs. Here the issue is that the test validates the exception message which is controlled by `DetailedErrors`. During the time we turned off e2etests, this option was modified to be true and therefore affected the exception message. The tests weren't updated in response to setting `DetailedErrors` to true and we saw failures when I re-enabled the tests. I opted to set `DetailedError` back to what it was before but the suggestion is that we can modify the exception message we verify to match the scenario where `DetailedError` is true as an alternative fix.

Comment (SteveSandersonMS) [reply src/Components/test/testassets/TestServer/ServerStartup.cs:22]: That makes sense! Thanks for explaining. All LGTM here then!

File: eng/targets/CSharp.Common.props [hunk 2]
```diff
@@ -12,7 +12,12 @@
     <PackageReference Include="MicroBuild.Core" Version="0.3.0" PrivateAssets="All" AllowExplicitReference="true" ExcludeAssets="All" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(UsingMicrosoftNETSdkWeb)' == 'true' OR '$(UsingMicrosoftNETSdkBlazorWebAssembly)' == 'true' OR '$(RazorSdkCurrentVersionProps)' != ''">
+  <PropertyGroup>
+    <_ReferenceLocalRazorSDK
```
Comment (pranavkm) [inline eng/targets/CSharp.Common.props:16]: @SteveSandersonMS this should fix the issue with you were seeing. I ran in to when debugging the tests in VS and really had to fix this to make any progression.

File: src/Components/test/E2ETest/Tests/WebAssemblyICUShardingTest.cs [hunk 3]
```diff
@@ -39,7 +36,7 @@ public void LoadingApp_FrenchLanguage_Works()
             Assert.Equal(culture.ToString(), cultureDisplay.Text);
 
             var dateDisplay = Browser.Exists(By.Id("dateTime"));
-            Assert.Equal(DisplayTime.ToString(culture), dateDisplay.Text);
+            Assert.Equal("02/09/2020 00:00:00", dateDisplay.Text);
```
Comment (pranavkm) [inline src/Components/test/E2ETest/Tests/WebAssemblyICUShardingTest.cs:39]: The ICU data on the server depends on the OS. On my machine, it's up to date with WASM but not so on the build machines. Using hardcoded strings seems like the best way to proceed: https://github.com/dotnet/runtime/pull/42136

**Comment by JunTaoLuo** [comment]
Thanks for the fix @pranavkm
